File: AppleDistribution.java
// https://leetcode.com/problems/apple-redistribution-into-boxes/description/?envType=daily-question&envId=2025-12-24

import java.util.Arrays;

public class AppleDistribution {
    static {
        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
            try (java.io.FileWriter fw = new java.io.FileWriter("display_runtime.txt")) {
                fw.write("Runtime = 0");
            } catch (Exception e) {
            }
        }));
    }
    public int minimumBoxes(int[] apple, int[] capacity) {
        int totalApples = 0;
        int count = 0;
        for(int i = 0; i < apple.length; i++){
            totalApples += apple[i];
        }
        Arrays.sort(capacity);
        for(int i = capacity.length - 1; i >= 0; i--){
            totalApples -= capacity[i];
            count++;
            if(totalApples <= 0){
                break;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        AppleDistribution solution = new AppleDistribution();
        int[] apple = {10, 20, 30};
        int[] capacity = {15, 25, 10, 5};
        int result = solution.minimumBoxes(apple, capacity);
        System.out.println("Minimum boxes needed: " + result);
    }
}

File: Python/AppleDistribution.py
# https://leetcode.com/problems/apple-redistribution-into-boxes/description/?envType=daily-question&envId=2025-12-24

from typing import List

class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        totalApples = sum(apple)

        capacity.sort(reverse = True)

        count = 0
        for i in capacity:
            totalApples -= i
            count += 1
            if totalApples <= 0:
                break
        
        return count
    
if __name__ == "__main__":
    solution = Solution()
    apple = [10, 20, 30]
    capacity = [15, 25, 10, 5]
    result = solution.minimumBoxes(apple, capacity)
    print("Minimum boxes needed:", result)  # Expected output: 3